packages/hierarchies-react/src/test/internal/TreeActions.test.ts

```diff
@@ -725,6 +731,48 @@ describe("TreeActions", () => {
   });
 
   describe("reloadTree", () => {
+    it("reloads nodes once when multiple request are made at the same time", async () => {
+      const model = createTreeModel([
+        {
+          id: undefined,
+          children: ["root-1", "root-2"],
+        },
+        {
+          id: "root-1",
+          isExpanded: true,
+          children: ["child-1"],
+        },
+        {
+          id: "root-2",
+          children: ["child-2"],
+        },
+      ]);
+
+      provider.getNodes.reset();
+      provider.getNodes.callsFake((props) => {
+        if (props.parentNode === undefined) {
+          return createAsyncIterator([createTestHierarchyNode({ id: "root-1" }), createTestHierarchyNode({ id: "root-2" })]);
+        }
+        if (HierarchyNodeKey.equals(props.parentNode.key, { type: "generic", id: "root-1" })) {
+          return createAsyncIterator([createTestHierarchyNode({ id: "child-1-2" })]);
+        }
+        return createAsyncIterator([]);
+      });
+
+      const actions = createActions(model);
+
+      actions.reloadTree({ state: "discard" });
+      actions.reloadTree({ state: "keep" });
+      actions.reloadTree({ state: "discard" });
+
+      await new Promise((r) => setTimeout(r, 50));
```

Comment:
this needs to go away - relying to specific timeout will result in a flaky test sooner or later

Reply:
The intention was to have the tests, before figuring out how to consistently check the results. Removed now.